cli/tests/pcluster/validators/test_ebs.py

```diff
@@ -123,3 +124,27 @@ def test_ebs_size_iops_validators(section_dict, expected_message):
 def test_ebs_volume_type_size_validator(section_dict, expected_message):
     actual_failures = SharedStorageSchema().load({"MountDir": "/my/mount/point", "EBS": section_dict}).validate()
     assert_failure_messages(actual_failures, expected_message)
+
+
+@pytest.mark.parametrize(
+    "volume_dict, expected_message",
+    [
+        (
+            {"Size": 25, "KmsKeyId": "arn:aws:kms:us-east-1:012345678910:1234abcd-12ab-34cd-56ef-1234567890ab"},
+            "KmsKeyId arn:aws:kms:us-east-1:012345678910:1234abcd-12ab-34cd-56ef-1234567890ab "
+            "is specified, the encrypted state can't be False.",
+        ),
+        (
+            {
+                "Size": 25,
+                "KmsKeyId": "arn:aws:kms:us-east-1:012345678910:1234abcd-12ab-34cd-56ef-1234567890ab",
+                "Encrypted": False,
+            },
+            "KmsKeyId arn:aws:kms:us-east-1:012345678910:1234abcd-12ab-34cd-56ef-1234567890ab "
+            "is specified, the encrypted state can't be False.",
+        ),
+    ],
+)
+def test_ebs_volume_kms_key_id_validator(volume_dict, expected_message):
+    actual_failures = VolumeSchema().load(volume_dict).validate()
```

Comment:
I would have more discussion on this. Since Marshmallow allows partial load, we don't have to load all required fields when testing. Moreover, because we are handling multiple error messages, we would like to make sure all validators in one schema/model work well together. I am trying to prove that with just a little sacrifice of the granularity of the unit tests, we are getting much more test coverage and better simulation of the real use case.

Reply:
Here I agree with @enrico-usai,  we still need to test Validator directly. For @hanwen-pcluste  case, can we put this in model unit tests for  handling multiple error messages? It would make sense that we test each unit in corresponding module.

Reply:
In unit tests individual _units or components_ of a software must be tested. The purpose is to validate that each unit of the code performs as expected, in this case we need to verify that the logic of the validator is the right one, according with the different values of the parameters.

If we put in the game marshmallow we are not just testing if the logic of the validator but we're also testing marshmallow loading, the schema definition, the partial loading of marshmallo, we're putting in the game the logic for implied and default values etc.

The simulation of the real case can be done by adding less granularity unit test like you're suggesting (or integration tests) but we cannot avoid the unit tests for the single validators.